### ISSUE TITLE
Results of the Social app survey

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/social#904]
The core results from the Social app survey are, in order of importance:
- Make Social compatible for 18 and 19 #844 #901 
- Limiting of visibility, private/public (just like Mastodon)
- Hashtag support
- Showing link embeds so they look nice
- Integration with Notifications
- Mobile apps are seen as important, and we can reference & advertise Mastodon apps here
- Recommendations for who to follow (colleagues, people who you worked with frequently, Nextcloud Mastodon account) so you have existing content and also see it as a useful work tool.
- Improved messaging/marketing around also using it as a work tool, not only for private

cc @daita @violoncelloCH @rullzer @jospoortvliet and @nextcloud/social :)

**Comment by StCyr** [comment]
Hi @jancborchardt 

May I ask where this survey comes from?

I'm asking because I believe some of the points are already solved:

- Limiting of visibility, private/public (just like Mastodon)
- Hashtag support, grouping

**Comment by jancborchardt** [comment]
Yes, sorry @StCyr this survey was running between May 27 – 9 June and got 159 responses.

Keep in mind the possibility of bias due to the audience of the survey. We primarily asked for people to fill it out via the [Nextcloud blog](https://nextcloud.com/blog/nextcloud-deck-social-and-forms-were-listening-to-your-thoughts-and-feature-wishes/) and the [@Nextclouders Twitter account](https://twitter.com/Nextclouders/status/1265589695293071360), which means that many respondents are likely to be people who know about Nextcloud, use it already, or are otherwise more technically apt.

But also, many people might not know about Social yet, so are not aware of what exists already. The questions were also to gauge how we are standing with our current features in comparison with other solutions, and what is important to polish. :)

Does that explain it?

**Comment by StCyr** [comment]
Hi @jancborchardt 

Thanks for your answer. It's a shame I didn't see that survey but anyway...

I just want to mention that I was hitting serious performance issues after using Social for a few months, with some DB queries taking up to 10 seconds or more (From memory I believe these were some `WHERE <column>='%foobar%'` queries on tables with tens of thousand of rows).

Are there any plans to review the DB 'structure by any chance?

PS: For the record, here's a link to an already solved issue that I've reported in the past regarding a DB structure issue: https://github.com/nextcloud/social/issues/737. I believe there are some other such issues still lurking around

**Comment by jancborchardt** [comment]
@StCyr oh ok – could you open a new issue for that database issue (or multiple ones if they are separate)? :)

**Comment by daita** [comment]
Hello @StCyr, glad to hear from you as I could not get any news from Telegram.
As you can see, the project is back on track.

Alpha3, that should be released today or tomorrow, include the first part of the fix: better structure.
The second part that is generating backup of oldest _streams_ yet needs to be done.

I hope you're ok, please contact me on Telegram !

**Comment by Mannshoch** [comment]
Any news on this APP?

**Comment by SchBenedikt** [comment]
> Any news on this APP?

@Mannshoch https://github.com/nextcloud/social/issues/1993; the new Version is working and will be published to the app store soon!

**Comment by SchBenedikt** [comment]
@Mannshoch 
New working release can be found here: 
https://apps.nextcloud.com/apps/social

**Comment by SchBenedikt** [comment]
Survey closed; ideas will be implemented soon/are implemented yet
